src/autolabel/data_loaders/read_datasets.py

```diff
@@ -0,0 +1,215 @@
+from typing import Dict, List, Union
+
+import logging
+import pandas as pd
+from dataclasses import dataclass
```

Comment:
let's replace this with pydantic for uniformity across the library? we already use pydantic for data type definitions in https://github.com/refuel-ai/autolabel/blob/main/src/autolabel/schema.py and a few other places -